src/features/session_trace/aggregate/index.js

```diff
@@ -1,538 +1,220 @@
-      return Object.values(partitionListByOriginMap).flat() // join the partitions back into 1-D, now ordered by origin then start time
-    }, this)
-    if (stns.length === 0) return {}
+    const hasReplay = this.agentRuntime.session?.state.sessionReplayMode === 1
+    const endUserId = this.agentInfo?.jsAttributes?.['enduser.id']
 
-    if (retry) {
-      this.sentTrace = this.trace
-    }
-    this.trace = {}
-    this.nodeCount = 0
-
-    let firstHarvestOfSession
-    if (this.agentRuntime.session) {
-      const isFirstPayload = !this.agentRuntime.session.state.traceHarvestStarted
-      firstHarvestOfSession = { fsh: Number(isFirstPayload) } // converted to '0' | '1'
-      if (isFirstPayload) this.agentRuntime.session.write({ traceHarvestStarted: true })
-    }
+    this.everHarvested = true
 
+    /** The blob consumer expects the following and will reject if not supplied:
+     * browser_monitoring_key
+     * type
+     * app_id
+     * protocol_version
+     * attributes
+     *
+     * For data that does not fit the schema of the above, it should be url-encoded and placed into `attributes`
+     */
     return {
       qs: {
-        st: this.agentRuntime.offset,
-        /** hr === "hasReplay" in NR1, standalone is always checked and processed before harvesting
-         * so a race condition between ST and SR states should not be a concern if implemented here */
-        hr: Number(!this.isStandalone),
-        /** fts === "firstTimestamp" in NR1, indicates what the earliest NODE timestamp was
-         * so that blob parsing doesn't need to happen to support UI/API functions  */
-        fts: this.agentRuntime.offset + earliestTimeStamp,
-        /** n === "nodeCount" in NR1, a count of nodes in the ST payload, so that blob parsing doesn't need to happen to support UI/API functions */
-        n: stns.length, // node count
-        ...firstHarvestOfSession
+        browser_monitoring_key: this.agentInfo.licenseKey,
+        type: 'BrowserSessionChunk',
+        app_id: this.agentInfo.applicationID,
+        protocol_version: '0',
+        attributes: encodeObj({
+          // this section of attributes must be controllable and stay below the query param padding limit -- see QUERY_PARAM_PADDING
+          // if not, data could be lost to truncation at time of sending, potentially breaking parsing / API behavior in NR1
+          // trace payload metadata
+          'trace.firstTimestamp': this.agentRuntime.offset + earliestTimeStamp,
+          'trace.lastTimestamp': this.agentRuntime.offset + latestTimeStamp,
+          'trace.nodes': stns.length,
+          'trace.originTimestamp': this.agentRuntime.offset,
+          // other payload metadata
+          agentVersion: this.agentRuntime.version,
+          ...(firstSessionHarvest && { firstSessionHarvest }),
+          ...(hasReplay && { hasReplay }),
+          ptid: `${this.agentRuntime.ptid}`,
+          rst: now(),
```

Comment:
Do we really still need rst with those 3 other timestamps?

Reply:
thanks, missed removing this